### PR TITLE
KIALI-3007 support watching CR in a different namespace

### PR DESCRIPTION
**Pull request by jmazzitelli**
https://issues.jboss.org/browse/KIALI-3007

I think this is going to be needed for https://issues.jboss.org/browse/KIALI-3007

NOTE: this PR does not affect current behavior. The defaults are set such that the behavior is left unchanged. This just allows someone to tell the install script to have the operator look elsewhere for the Kiali CR (as opposed to the same namespace where the operator is located). The default is the operator will look in its own namespace (which is how it works today).

### CODE REVIEW

**Comment by jmazzitelli** [comment]
The merge-operator-yaml.sh is here because I think we are going to need this when we have to build the yaml to give to Maistra. Rather than have to manually build/change the single yaml for Maistra to use, I think we can use this script to build it for us. I'm still not 100% sure it is going to be used, but I put it in here because I think it might used.

**Comment by jmazzitelli** [comment]
Here's how to test this.

First, make sure you build everything from the top directory: `make docker-build`.

The remaining steps I will assume your current working directory is `operator/deploy`.

Now test 2 things - one using SingleNamespace mode with namespace-scoped operator. Then AllNamespace mode with cluster-scoped operator.

## TEST 1

Test SingleNamespace mode, namespace-scoped operator

Delete everything to make sure you start from scratch: `./deploy-kiali-operator.sh -um true`

Run the deploy script and tell it you want to change the operator watch namespace to a specific namespace that is NOT kiali-operator. Let's test with `myproject` namespace.

`./deploy-kiali-operator.sh -oiv dev -own myproject`

Answer the prompts. This will now install the Kiali CR in the `myproject` namespace - confirm that:

`oc get kiali kiali -n myproject`

Confirm also that Kiali is installed in istio-system namespace:

`oc get deployment kiali -n istio-system`

## TEST 2

Test AllNamespace mode, cluster-scoped operator

Delete everything to make sure you start from scratch: `./deploy-kiali-operator.sh -um true`

Run the deploy script and tell it you want to change the operator watch namespace to all namespaces. That means setting the watch namespace to `**`. Note that because the script won't know where to install Kiali, it will not attempt to do so (we'll install Kiali in the next step):

`./deploy-kiali-operator.sh -oiv dev -own '**'`

Now use the Makefile to create a Kiali CR in some custom namespace - we'll create a namespace called "foo" and put it in there:

```
oc create namespace foo
OPERATOR_WATCH_NAMESPACE=foo make -C .. kiali-create
```

Confirm the Makefile put the Kiali CR in the `foo` namespace - confirm that:

`oc get kiali kiali -n foo`

After you give the operator some time to complete everything, confirm that Kiali is installed in istio-system namespace:

`oc get deployment kiali -n istio-system`

File: operator/deploy/deploy-kiali-operator.sh [hunk 1]
```diff
@@ -797,27 +818,41 @@ fi
...
+  local _ns="${OPERATOR_WATCH_NAMESPACE}"
+  if [ "${_ns}" == '""' ]; then
+    _ns="<any namespace you choose>"
   fi
   echo "=========================================="
   echo "Skipping the automatic Kiali installation."
-  echo "To install Kiali, create a Kiali custom resource in the namespace [${OPERATOR_NAMESPACE}]."
+  echo "To install Kiali, create a Kiali custom resource in the namespace [$_ns]."
   echo "An example Kiali CR with all settings documented can be found here:"
   echo "  https://raw.githubusercontent.com/kiali/kiali/${_BRANCH}/operator/deploy/kiali/kiali_cr.yaml"
```
Comment (jshaughn) [inline operator/deploy/deploy-kiali-operator.sh]: This URL is bad. Use:
https://github.com/kiali/kiali/blob/${_BRANCH}/master/operator/deploy/kiali/kiali_cr.yaml

Also, below.

**Comment by jmazzitelli** [comment]
This is needed for the Maistra work that is starting: https://issues.jboss.org/browse/MAISTRA-369 so I'm going to merge.

**Pull request merged**